cmd/source_profile.go

```diff
@@ -86,58 +105,86 @@ type SourceProfileConnectionPostgreSQL struct {
 	pwd  string // Same as PGPASSWORD environment variable
 }
 
-func NewSourceProfileConnectionPostgreSQL(params map[string]string) SourceProfileConnectionPostgreSQL {
+func NewSourceProfileConnectionPostgreSQL(params map[string]string) (SourceProfileConnectionPostgreSQL, error) {
 	pg := SourceProfileConnectionPostgreSQL{}
-	if host, ok := params["host"]; ok {
-		pg.host = host
-	}
-	if port, ok := params["port"]; ok {
-		pg.port = port
-	} else { // Set default port for postgresql, which rarely changes.
-		pg.port = "5432"
+	host, hostOk := params["host"]
+	user, userOk := params["user"]
+	db, dbOk := params["db_name"]
+	port, portOk := params["port"]
+	pwd, pwdOk := params["password"]
+	if !(hostOk || userOk || dbOk || portOk || pwdOk) {
+		// No connection params provided through source-profile. Fetching from env variables.
+		fmt.Printf("Connection parameters not specified in source-profile. Reading from " +
+			"environment variables PGHOST, PGUSER, PGDATABASE, PGPORT, PGPASSWORD...\n")
+		pg.host = os.Getenv("PGHOST")
+		pg.user = os.Getenv("PGUSER")
+		pg.db = os.Getenv("PGDATABASE")
+		pg.port = os.Getenv("PGPORT")
+		pg.pwd = os.Getenv("PGPASSWORD")
+	} else if hostOk && userOk && dbOk {
+		// All connection params provided through source-profile. Port and password handled later.
+		pg.host, pg.user, pg.db, pg.port, pg.pwd = host, user, db, port, pwd
+	} else {
+		return pg, fmt.Errorf("please specify host, port, user and db_name in the source-profile")
 	}
-	if user, ok := params["user"]; ok {
-		pg.user = user
+
+	// Throw same error if the input entered is empty.
+	if pg.host == "" || pg.user == "" || pg.db == "" {
+		return pg, fmt.Errorf("found empty string for host/user/db. please specify host, port, user and db_name in the source-profile")
 	}
-	if db, ok := params["db_name"]; ok {
-		pg.db = db
+
+	if pg.port == "" {
+		// Set default port for postgresql, which rarely changes.
+		pg.port = "5432"
 	}
-	if pwd, ok := params["password"]; ok {
-		pg.pwd = pwd
+	if pg.pwd == "" {
+		pg.pwd = conversion.GetPassword()
 	}
-	return pg
+
+	return pg, nil
 }
 
 type SourceProfileConnectionDynamoDB struct {
-	awsAccessKeyID     string // Same as AWS_ACCESS_KEY_ID environment variable
-	awsSecretAccessKey string // Same as AWS_SECRET_ACCESS_KEY environment variable
-	awsRegion          string // Same as AWS_REGION environment variable
-	dydbEndpoint       string // Same as DYNAMODB_ENDPOINT_OVERRIDE environment variable
-	schemaSampleSize   int64  // Number of rows to use for inferring schema (default 100,000)
+	// We don't need any connection parameters here as SDK reads them from the env variables.
```

Comment:
I don't see any harm in keeping the parsed flags for later reference either. It's similar to what we are doing for SourceProfileConnectionPostgreSQL and SourceProfileConnectionMySQL - there too we could just store `sqlConnectionStr` instead of storing all of host, port, user, db etc. In fact it takes more effort for a reader to understand the comment instead.

A case in point would be when we eventually move to passing around SourceProfile to SchemaConv and DataConv, then source database specific code such as setting environment variables for DynamoDB would move inside sources/dynamodb/.